iris-mpc-cpu/src/execution/local.rs

```diff
@@ -17,6 +17,18 @@ pub fn generate_local_identities() -> Vec<Identity> {
     ]
 }
 
+fn get_free_port() -> eyre::Result<u16> {
+    let listener = std::net::TcpListener::bind("127.0.0.1:0")?;
+    Ok(listener.local_addr()?.port())
+}
+
+pub fn get_free_local_addresses(num_ports: usize) -> eyre::Result<Vec<String>> {
+    let free_ports = (0..num_ports)
+        .map(|_| get_free_port())
+        .collect::<eyre::Result<Vec<u16>>>()?;
+    Ok(free_ports.iter().map(|p| format!("[::1]:{p}")).collect())
+}
+
```

Comment:
This does not seem to be very robust.
First, it drops the listener after getting its port, so it could be immediately reused, second it grabs ports on IPv4 and returns IPv6 address strings?

Reply:
I added a static variable to keep track of used ports. Please, have a look.

Reply:
Oh I see, this is only intended to be used for testing?

Reply:
Yes